project-addons/crm_claim_rma_custom/wizard/crm_phonecall.py

```diff
@@ -40,11 +41,30 @@ class crm_phonecall(models.Model):
     """ Wizard for CRM phonecalls"""
     _inherit = "crm.phonecall"
 
-    partner_id = fields.Many2one('res.partner', 'Contact', required=True)
+    local_tz = pytz.timezone('Europe/Madrid')
+
+    name = fields.Char('Call Summary', readonly=True)
+    partner_id = fields.Many2one('res.partner', 'Contact', required=True,
+                                 domain="[['is_company', '=', 1], ['customer', '=', True]]")
     start_date = fields.Datetime('Start Date', readonly=True, default=fields.Datetime.now)
-    name = fields.Text('Call Summary')
     user_id = fields.Many2one('res.users', 'Responsible', readonly=True)
     call_type = fields.Selection(CALL_TYPE,'Call type', required=True)
+    description = fields.Text('Call Description')
+
+    def utc_to_local(self, utc_dt):
+        local_dt = utc_dt.replace(tzinfo=pytz.utc).astimezone(self.local_tz)
+        return self.local_tz.normalize(local_dt)
+
+    @api.onchange('partner_id')
+    def _calc_name(self):
```

Comment:
Esto debería hacerse en el create, y así no dependéis de un on_change, si se crea la llamada desde código no se llama al on_change nunca.